contracts/wallet.sol

```diff
@@ -534,13 +372,18 @@ contract Vault is AddressWhitelist, SpendLimit, ERC165, Transferrable, Balanceab
     /// @dev this is an internal nonce to prevent replay attacks from relayer
     uint public relayNonce;
 
-    /// @dev Constructor initializes the vault with an owner address and spend limit. It also sets up the controllable and tokenWhitelist contracts with the right name registered in ENS.
+    /// @dev Constructor initializes the vault with an owner address and daily limit. It also sets up the controllable and tokenWhitelist contracts with the right name registered in ENS.
     /// @param _owner_ is the owner account of the wallet contract.
     /// @param _transferable_ indicates whether the contract ownership can be transferred.
     /// @param _tokenWhitelistNode_ is the ENS node of the Token whitelist.
     /// @param _controllerNode_ is the ENS name node of the controller.
-    /// @param _spendLimit_ is the initial spend limit.
-    constructor(address payable _owner_, bool _transferable_, bytes32 _tokenWhitelistNode_, bytes32 _controllerNode_, uint _spendLimit_) SpendLimit(_spendLimit_) Ownable(_owner_, _transferable_) Controllable(_controllerNode_) TokenWhitelistable(_tokenWhitelistNode_) public {}
+    /// @param _dailyLimit_ is the initial daily limit in stablecoin units e.g. 1K USD.
+    constructor(address payable _owner_, bool _transferable_, bytes32 _tokenWhitelistNode_, bytes32 _controllerNode_, uint _dailyLimit_) Ownable(_owner_, _transferable_) Controllable(_controllerNode_) TokenWhitelistable(_tokenWhitelistNode_) public {
+        // Get the stablecoin's magnitude.
+       ( ,uint256 stablecoinMagnitude, , , , , ) = _getStablecoinInfo();
+       require(stablecoinMagnitude > 0, "no stablecoin");
+       _initializeDailyLimit(_dailyLimit_ * stablecoinMagnitude);
```

Comment:
it's inconsistent what we're doing here - this is the only parameter where daily limit is passed as the base unit value

Reply:
True, but it is the only way if we don't know the stablecoin in advance or we want to somehow update it, DAI: 18 decimals, USDC: 6 decimals, GUSD: 2 decimals etc.

Reply:
I suppose you could move the _getStableCoinInfo() call inside the _initializeDailyLimit() method and pass only the number of stablecoins e.g. 100 DAI instead of muliplying, and if we want to pass a different stablecoin to the constructor you would just pass a different stablecoin address as a perameter.

Reply:
Come to think of it @i-stam - the stablecoin is hardcoded in the tokenWhitelist, I think it would be better to be able to set the stablecoin address by one of our controllers. Otherwise if something goes wrong with the stablecoin contract we can change it to a different one.

Reply:
This is true, we never really discussed about it, let's do it in the meeting.